src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -40,7 +45,39 @@
      * София - 900</p>
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws Exception, IllegalArgumentException {
+        DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateStart = LocalDate.parse(dateFrom, dateFormat);
+        LocalDate dateEnd = LocalDate.parse(dateTo, dateFormat);
+
+        if (dateStart.isAfter(dateEnd)) {
+            throw new IllegalDateParametersException();
+        }
+
+        StringBuilder report = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo);
+
+        for (String workerName :
+                names) {
+            int sumOfMoney = 0;
+            for (String dataLine :
+                    data) {
+                if (dataLine.contains(workerName)) {
+                    String[] tempData = dataLine.split(" ");
+                    LocalDate tempStart = LocalDate.parse(tempData[0], dateFormat);
+                    if (isaCorrectDate(dateStart, dateEnd, tempStart)) {
+                        sumOfMoney += Integer.parseInt(tempData[2]) * Integer.parseInt(tempData[3]);
+                    }
+                }
+            }
+            report.append("\n").append(workerName).append(" - ").append(sumOfMoney);
+        }
+        return String.valueOf(report.append("\n"));
+    }
+
+    private boolean isaCorrectDate(LocalDate dateStart, LocalDate dateEnd, LocalDate tempStart) {
```

Comment:
this method looks hard. think about better naming and use only two parameters in a signature